internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -247,8 +248,37 @@ func resourceRecoveryServicesVaultCreateUpdate(d *pluginsdk.ResourceData, meta i
 		},
 	}
 
-	if _, err = storageCfgsClient.Update(ctx, id.Name, id.ResourceGroup, storageCfg); err != nil {
-		return fmt.Errorf("updating Recovery Service Storage Cfg %s: %+v", id.String(), err)
+	err = pluginsdk.Retry(stateConf.Timeout, func() *pluginsdk.RetryError {
+		if resp, err := storageCfgsClient.Update(ctx, id.Name, id.ResourceGroup, storageCfg); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return pluginsdk.RetryableError(fmt.Errorf("updating Recovery Service Storage Cfg %s: %+v", id.String(), err))
+			}
+			if resp.Response.StatusCode == 400 { // ResourceNotYetSynced error
+				return pluginsdk.RetryableError(fmt.Errorf("updating Recovery Service Storage Cfg %s: %+v", id.String(), err))
+			}
+
+			return pluginsdk.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	// storage type is not updated instantaneously, so we wait until storage type is correct
+	err = pluginsdk.Retry(stateConf.Timeout, func() *pluginsdk.RetryError {
+		if resp, err := storageCfgsClient.Get(ctx, id.Name, id.ResourceGroup); err == nil {
+			if resp.Properties.StorageType != storageCfg.Properties.StorageModelType {
+				return pluginsdk.RetryableError(fmt.Errorf("updating Recovery Service Storage Cfg %s: %+v", id.String(), err))
```

Comment:
the `id` is wrapped below, so we can remove this:

```suggestion
				return pluginsdk.RetryableError(fmt.Errorf("updating Storage Config: %+v", err))
```